plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -816,39 +949,69 @@ public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessio
                 .map(ignored -> singletonList(prefix.toSchemaTableName()))
                 .orElseGet(() -> listTables(session, prefix.getSchema()));
 
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+
         return tables.stream()
-                .flatMap(table -> {
+                .flatMap(tableName -> {
                     try {
-                        if (redirectTable(session, table).isPresent()) {
+                        if (redirectTable(session, tableName).isPresent()) {
                             // put "redirect marker" for current table
-                            return Stream.of(TableColumnsMetadata.forRedirectedTable(table));
+                            return Stream.of(TableColumnsMetadata.forRedirectedTable(tableName));
                         }
 
-                        Optional<DeltaMetastoreTable> metastoreTable = metastore.getTable(table.getSchemaName(), table.getTableName());
+                        Optional<Table> metastoreTable = metastore.getRawMetastoreTable(tableName.getSchemaName(), tableName.getTableName());
                         if (metastoreTable.isEmpty()) {
                             // this may happen when table is being deleted concurrently,
                             return Stream.of();
                         }
-                        String tableLocation = metastoreTable.get().location();
-                        TableSnapshot snapshot = transactionLogAccess.loadSnapshot(session, table, tableLocation, Optional.empty());
+
+                        Table table = metastoreTable.get();
+                        verifyDeltaLakeTable(table);
+
+                        String tableLocation = HiveMetastoreBackedDeltaLakeMetastore.getTableLocation(table);
+                        if (cacheTableMetadata && table.getParameters().containsKey(TRINO_LAST_TRANSACTION_VERSION) && table.getParameters().containsKey(TRINO_METADATA_SCHEMA_STRING)) {
+                            long version = Long.parseLong(table.getParameters().get(TRINO_LAST_TRANSACTION_VERSION));
+                            if (isLatestVersion(fileSystem, tableLocation, version)) {
```

Comment:
Could we extract method for checking if tableHandle is present in cache? It would be more clear what we are doing here.
It's also very similar to check from `streamRelationComments`.